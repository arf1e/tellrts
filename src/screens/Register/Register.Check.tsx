import React, {useState} from 'react';
import FullScreenModal from '../../components/Modals';
import {ActivityIndicator, View} from 'react-native';

import styles from './Register.styles';
import Field from '../../components/Field';
import {REGISTER_FORM_VALUES} from './Register.types';
import {FormikProps} from 'formik';
import {BodyCopy} from '../../components/Typography';
import Link, {ImageLink} from '../../components/Links';
import {useTranslation} from 'react-i18next';
import colors from '../../utils/colors';
import PrimaryButton from '../../components/Buttons';
import {
  BIRTHDAY_STEP,
  FORM_STEP,
  getStepIndex,
  LOCATION_STEP,
  NAME_STEP,
  PHOTO_STEP,
  SEX_STEP,
} from './Register.utils';

type Props = {
  formikProps: FormikProps<REGISTER_FORM_VALUES>;
  navigateForm: (index: number) => void;
  registerLoading?: boolean;
  submitError?: any;
};

type CheckFieldProps = {
  title: string;
  value: string;
  onValuePress: () => void;
};

const CheckTextField = ({title, value, onValuePress}: CheckFieldProps) => {
  const {t} = useTranslation();
  return (
    <View style={styles.checkTextField}>
      <BodyCopy style={styles.checkTextFieldTitle}>{title}</BodyCopy>
      <Link onPress={onValuePress}>{value || t('register.check.noValue')}</Link>
    </View>
  );
};

const Check = ({formikProps, navigateForm, registerLoading}: Props) => {
  const [isEmailModalActive, setIsEmailModalActive] = useState(false);
  const {t} = useTranslation();
  const getToFormStep = (step: FORM_STEP) => {
    const index = getStepIndex(step);
    navigateForm(index);
  };
  return (
    <View>
      {registerLoading ? (
        <ActivityIndicator
          size="large"
          color={colors.secondary}
          testID="Register.Check.Loading"
        />
      ) : (
        <View style={styles.checkFields}>
          <ImageLink
            onPress={() => getToFormStep(PHOTO_STEP)}
            size={150}
            uri={formikProps.values.photo.path}
          />
          <CheckTextField
            title={t('register.check.nameTitle')}
            value={formikProps.values.name}
            onValuePress={() => getToFormStep(NAME_STEP)}
          />
          <CheckTextField
            title={t('register.check.dobTitle')}
            value={formikProps.values.birthdayInput}
            onValuePress={() => getToFormStep(BIRTHDAY_STEP)}
          />
          <CheckTextField
            title={t('register.check.sexTitle')}
            value={
              formikProps.values.sex
                ? t('register.sex.male')
                : t('register.sex.female')
            }
            onValuePress={() => getToFormStep(SEX_STEP)}
          />
          <CheckTextField
            title={t('register.check.cityTitle')}
            value={formikProps.values.cityTitle}
            onValuePress={() => getToFormStep(LOCATION_STEP)}
          />
          <CheckTextField
            title={t('register.check.emailTitle')}
            value={formikProps.values.email}
            onValuePress={() => setIsEmailModalActive(true)}
          />
        </View>
      )}
      <FullScreenModal
        active={isEmailModalActive}
        closeModal={() => setIsEmailModalActive(false)}
        title={t('register.check.emailTitle')}>
        <View>
          <BodyCopy style={styles.fieldTitle}>
            {t('register.check.emailFieldTitle')}
          </BodyCopy>
          <Field
            value={formikProps.values.email}
            onChangeText={formikProps.handleChange('email')}
            placeholder={t('register.check.emailFieldPlaceholder')}
            textContentType="emailAddress"
            autoCapitalize="none"
            autoFocus
          />
          <PrimaryButton
            style={styles.emailModalFieldButton}
            title={t('register.check.emailApplyButton')}
            disabled={Boolean(formikProps.errors.email)}
            onPress={() => setIsEmailModalActive(false)}
          />
        </View>
      </FullScreenModal>
    </View>
  );
};

export default Check;
