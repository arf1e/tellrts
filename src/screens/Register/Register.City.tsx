import React, {useState} from 'react';
import {useTranslation} from 'react-i18next';
import {ActivityIndicator, Pressable, View} from 'react-native';
import {GoogleAutoComplete} from 'react-native-google-autocomplete';

import FullScreenModal from '../../components/Modals';
import Link from '../../components/Links';
import {BodyCopy, Subtitle} from '../../components/Typography';

import styles from './Register.styles';
import Field from '../../components/Field';
import colors from '../../utils/colors';
import {GoogleLocationResult} from 'react-native-google-autocomplete/dist/services/Google.service';
import PrimaryButton from '../../components/Buttons';

const GOOGLE_API_KEY = 'AIzaSyDw6jMkY0hQFicfxsmzU1bKn-sFAHWUgS0';

type Props = {
  countrySelected: string | null;
  setCity: (id: string, title: string) => void;
  currentCity: {
    id: string;
    title: string;
  };
};

type CityResultProps = {
  city: GoogleLocationResult;
  onPress: () => void;
};

const CityResult = ({city, onPress}: CityResultProps) => {
  const btnStyles = [styles.optionContainer, styles.optionContainerCity];
  return (
    <Pressable style={btnStyles} onPress={onPress}>
      <BodyCopy style={styles.optionTitle}>
        {city.structured_formatting.main_text}
      </BodyCopy>
      <BodyCopy style={styles.cityDescription}>
        {city.structured_formatting.secondary_text}
      </BodyCopy>
    </Pressable>
  );
};

const City = ({countrySelected, setCity, currentCity}: Props) => {
  const {t} = useTranslation();
  const [modalActive, setModalActive] = useState(false);
  const [cityValue, setCityValue] = useState(currentCity.title || '');

  const handleChooseCity = (
    id: string,
    title: string,
    clearSearch: () => void,
  ) => {
    setCity(id, title);
    setCityValue(title);
    clearSearch();
  };

  const renderLinkTitle = () => {
    if (!currentCity.title) {
      return t('register.location.cityButton');
    } else {
      return currentCity.title;
    }
  };

  const handleLocationResults = (
    locationResults: GoogleLocationResult[],
    clearSearch: () => void,
  ): React.ReactNode => {
    const threeClosestResults = locationResults.slice(0, 3);
    return threeClosestResults.map(el => (
      <CityResult
        key={el.place_id}
        onPress={() =>
          handleChooseCity(
            el.place_id,
            el.structured_formatting.main_text,
            clearSearch,
          )
        }
        city={el}
      />
    ));
  };

  return (
    <View style={styles.cityContainer}>
      <Subtitle>{t('register.location.cityTitle')}</Subtitle>
      <Link onPress={() => setModalActive(true)}>{renderLinkTitle()}</Link>
      <FullScreenModal
        closeModal={() => setModalActive(false)}
        title={t('register.location.cityButton')}
        active={modalActive}>
        <GoogleAutoComplete
          apiKey={GOOGLE_API_KEY}
          debounce={300}
          minLength={2}
          // @ts-ignore
          components={`country:${countrySelected.toLowerCase()}`}
          queryTypes="(cities)">
          {({handleTextChange, locationResults, clearSearch, isSearching}) => (
            <>
              <Field
                value={cityValue}
                autoFocus
                placeholder={t('register.location.cityInputPlaceholder')}
                onChangeText={(text: string) => {
                  setCityValue(text);
                  if (text.length >= 2) {
                    handleTextChange(text);
                  } else {
                    clearSearch();
                  }
                }}
              />
              <View>
                {isSearching && cityValue.length > 2 && (
                  <ActivityIndicator size="small" color={colors.primary} />
                )}
                {Boolean(locationResults) &&
                  handleLocationResults(locationResults, clearSearch)}
                {Boolean(currentCity.id) && (
                  <PrimaryButton
                    style={styles.cityApplyBtn}
                    title={t('register.location.apply')}
                    onPress={() => setModalActive(false)}
                  />
                )}
              </View>
            </>
          )}
        </GoogleAutoComplete>
      </FullScreenModal>
    </View>
  );
};

export default City;
