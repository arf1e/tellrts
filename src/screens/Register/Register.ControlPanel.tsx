import React from 'react';
import {useTranslation} from 'react-i18next';
import {View} from 'react-native';

import PrimaryButton, {SecondaryButton} from '../../components/Buttons';
import styles from './Register.styles';

interface Props {
  primaryAction: () => any;
  secondaryAction: () => any;
  style?: any;
  disabled?: boolean;
}

export default ({
  primaryAction,
  secondaryAction,
  style,
  disabled,
}: Props): React.ReactElement => {
  const {t} = useTranslation();
  return (
    <View style={[styles.controlPanel, style ? style : null]}>
      <SecondaryButton
        title={t('register.controls.back')}
        onPress={secondaryAction}
        style={styles.utilSecondary}
        testID="Register.ControlPanel.back"
      />
      <PrimaryButton
        title={t('register.controls.next')}
        onPress={primaryAction}
        style={styles.utilPrimary}
        disabled={disabled}
        testID="Register.ControlPanel.next"
      />
    </View>
  );
};
