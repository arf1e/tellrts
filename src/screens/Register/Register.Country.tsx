import React, {useEffect, useState} from 'react';
import {View, Pressable} from 'react-native';
import Field from '../../components/Field';
import Reanimated, {
  FadeIn,
  FadeOut,
  interpolate,
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';
import Emoji from 'react-native-emoji';

import FullScreenModal from '../../components/Modals';
import {BodyCopy} from '../../components/Typography';
import {
  getCountriesByName,
  getCountryNameByCode,
  Country as CountryType,
} from '../../components/Modals/countries';
import styles from './Register.styles';
import PrimaryButton from '../../components/Buttons';
import {I18nContext, useTranslation} from 'react-i18next';
import Link from '../../components/Links/Link';
import i18next from 'i18next';

const AnimatedPressable = Reanimated.createAnimatedComponent(Pressable);

type CountryProps = {
  setCountryCode: (code: string) => void;
  countryTitle?: string;
  countrySelected: string | null;
  setcountryTitle: (name: string) => void;
  clearCitySelection: () => void;
};

const CountryItem = ({
  country,
  onPress,
}: {
  country: CountryType;
  onPress: () => void;
}) => {
  const language: 'ru' | 'en' = i18next.language;
  return (
    <AnimatedPressable
      entering={FadeIn.duration(200)}
      exiting={FadeOut.duration(150)}
      style={styles.optionContainer}
      onPress={onPress}>
      <Emoji name={`flag-${country.code.toLowerCase()}`} style={styles.flag} />
      <BodyCopy style={styles.optionTitle}>{country.name[language]}</BodyCopy>
    </AnimatedPressable>
  );
};

const Country = ({
  setCountryCode,
  countryTitle,
  setcountryTitle,
  countrySelected,
  clearCitySelection,
}: CountryProps) => {
  const [isModalActive, setIsModalActive] = useState(false);
  const [countriesResult, setCountriesResult] = useState<CountryType[]>([]);
  const {t} = useTranslation();
  const language: 'ru' | 'en' = i18next.language;

  const handleCountryNameInput = (countryName: string) => {
    setCountryCode('');
    setcountryTitle(countryName);
    const matchedCountries = getCountriesByName(countryName).slice(0, 4);
    setCountriesResult(matchedCountries);
  };

  const handleCountryItemSelection = (country: CountryType) => {
    clearCitySelection();
    setcountryTitle(country.name[language]);
    setCountryCode(country.code);
    setCountriesResult([]);
  };

  const renderCountriesResult = () => (
    <View style={styles.resultsContainer}>
      {countriesResult.map(country => (
        <CountryItem
          country={country}
          onPress={() => handleCountryItemSelection(country)}
          key={country.code}
        />
      ))}
    </View>
  );

  const renderConfirmButton = () => (
    <PrimaryButton
      onPress={() => setIsModalActive(false)}
      title={t('register.location.apply')}
      style={styles.countryConfirmBtn}>
      <Emoji
        // @ts-ignore
        name={`flag-${countrySelected.toLowerCase()}`}
        style={styles.countryConfirmBtnEmoji}
      />
    </PrimaryButton>
  );

  const renderLinkTitle = (): string => {
    if (!countrySelected) {
      return t('register.location.countryPickModal');
    }
    return getCountryNameByCode(countrySelected);
  };

  return (
    <View>
      <Link onPress={() => setIsModalActive(true)}>{renderLinkTitle()}</Link>
      <FullScreenModal
        active={isModalActive}
        closeModal={() => setIsModalActive(false)}
        title={t('register.location.countryPickModal')}>
        <Field
          placeholder="Start typing..."
          onChangeText={handleCountryNameInput}
          style={styles.countryTitle}
          value={countryTitle}
          scrollEnabled={false}
        />
        {countriesResult.length > 0 && renderCountriesResult()}
        {Boolean(countrySelected) && renderConfirmButton()}
      </FullScreenModal>
    </View>
  );
};

export default Country;
