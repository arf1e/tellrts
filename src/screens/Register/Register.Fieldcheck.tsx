import React, {useEffect} from 'react';
import {Text, View} from 'react-native';
import Reanimated, {
  interpolateColor,
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';

import animationConstants from '../../utils/animationConstants';
import colors from '../../utils/colors';

import styles from './Register.styles';

type Props = {
  isPassing: boolean;
  title: string;
};

const AnimatedDot = Reanimated.createAnimatedComponent(View);
const AnimatedText = Reanimated.createAnimatedComponent(Text);

const Fieldcheck = ({isPassing, title}: Props) => {
  const colorChangeShared = useSharedValue(0);

  const animatedDotStyle = useAnimatedStyle(
    () => ({
      backgroundColor: interpolateColor(
        colorChangeShared.value,
        [0, 1],
        [colors.lightGray, colors.good],
      ),
    }),
    [colorChangeShared],
  );

  const animatedTextStyle = useAnimatedStyle(
    () => ({
      color: interpolateColor(
        colorChangeShared.value,
        [0, 1],
        [colors.gray, colors.good],
      ),
    }),
    [colorChangeShared],
  );

  const dotStyle = [styles.fieldcheckDot, animatedDotStyle];
  const textStyle = [styles.fieldcheckText, animatedTextStyle];

  useEffect(() => {
    if (isPassing) {
      colorChangeShared.value = withTiming(1, {
        duration: animationConstants.BUTTON_IN,
      });
    } else {
      colorChangeShared.value = withTiming(0, {
        duration: animationConstants.BUTTON_OUT,
      });
    }
  }, [isPassing, colorChangeShared]);

  return (
    <View style={styles.fieldcheckContainer}>
      <AnimatedDot style={dotStyle} />
      <AnimatedText style={textStyle}>{title}</AnimatedText>
    </View>
  );
};

export default Fieldcheck;
