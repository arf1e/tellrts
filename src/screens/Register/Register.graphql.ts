import {gql} from '@apollo/client';

export const REGISTER_MUTATION = gql`
  mutation createAccount(
    $email: String!
    $name: String!
    $birthday: String!
    $password: String!
    $sex: Boolean!
    $countryCode: String
    $photo: Upload
    $cityId: String
  ) {
    createAccount(
      name: $name
      birthday: $birthday
      sex: $sex
      photo: $photo
      countryCode: $countryCode
      cityId: $cityId
      password: $password
      email: $email
    ) {
      ok
      error
      token
    }
  }
`;
