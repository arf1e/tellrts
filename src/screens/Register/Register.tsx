import {useBackHandler} from '@react-native-community/hooks';
import {ParamListBase} from '@react-navigation/native';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import React, {useState} from 'react';
import {Alert, SafeAreaView, StatusBar, View} from 'react-native';

import styles from './Register.styles';
import colors from '../../utils/colors';

import DismissKeyboard from '../../components/DismissKeyboard';
import {LOGIN_SCREEN} from '../../components/Navigation/AuthNavigator';
import {Formik} from 'formik';
import {
  validateSchema,
  schema,
  REGISTER_FORM_INITIAL_VALUES,
} from './Register.utils';
import {useTranslation} from 'react-i18next';
import {useMutation} from '@apollo/client';
import errorCatcher from '../../utils/toasts';
import {generateRNFile} from '../../utils/photo';
import {logIn} from '../../utils/slices/authSlice';
import {useDispatch} from 'react-redux';
import {REGISTER_MUTATION} from './Register.graphql';
import {GraphQLError} from 'graphql';
import {REGISTER_FORM_VALUES} from './Register.types';
import RegisterFormNavigation from './Register.FormNavigation';

const Register = ({
  navigation,
  route,
}: NativeStackScreenProps<ParamListBase>) => {
  const [step, setStep] = useState(0);
  const nextStep = () => setStep(step + 1);
  const dispatch = useDispatch();
  const previousStep = () => {
    if (step === 0) {
      askIfGetBackToLoginPage();
      return false;
    } else {
      setStep(step - 1);
      return true;
    }
  };
  const [isReviewing, setIsReviewing] = useState(false);

  const {t} = useTranslation();
  const getToReviewStep = () => setStep(6);

  const navigateFormDuringReview = (index: number) => {
    setIsReviewing(true);
    setStep(index);
  };

  const askIfGetBackToLoginPage = () =>
    Alert.alert(t('register.quit.title'), t('register.quit.description'), [
      {
        text: 'Cancel',
        style: 'cancel',
        onPress: () => null,
      },
      {
        text: 'Confirm',
        style: 'destructive',
        onPress: () => navigation.navigate(LOGIN_SCREEN),
      },
    ]);

  useBackHandler(() => {
    if (step > 0) {
      setStep(step - 1);
      return true;
    } else {
      askIfGetBackToLoginPage();
      return true;
    }
  });

  const onRegisterMutationCompleted = (
    {error, token}: {error: GraphQLError; token: string},
    reset: () => void,
  ) => {
    if (error) {
      errorCatcher({
        name: t('register.check.registerError'),
        message: error,
      });
      console.log(error);
      reset();
      return;
    }

    dispatch(logIn({token}));
  };
  // @ts-ignore
  const [register, {loading: registerLoading, reset: registerReset}] =
    useMutation(REGISTER_MUTATION, {
      // @ts-ignore
      onCompleted: data =>
        onRegisterMutationCompleted(data.createAccount, registerReset),
      onError: e => {
        errorCatcher(e);
        registerReset();
      },
    });

  const handleRegister = async (values: REGISTER_FORM_VALUES) => {
    const {birthdayInput} = values;
    const photoFile = await generateRNFile(values.photo);
    await register({
      variables: {
        name: values.name,
        email: values.email,
        birthday: birthdayInput,
        sex: values.sex,
        photo: photoFile,
        countryCode: values.countryCode,
        cityId: values.cityId,
        password: values.password,
      },
      fetchPolicy: 'no-cache',
    });
  };

  return (
    <DismissKeyboard>
      <SafeAreaView style={styles.screenWrapper}>
        <StatusBar
          backgroundColor={colors.background}
          barStyle="dark-content"
        />
        <View style={styles.container}>
          <Formik
            initialValues={{
              ...REGISTER_FORM_INITIAL_VALUES,
              // @ts-ignore
              email: route.params?.email || '',
            }}
            onSubmit={values => console.log(values)}
            validate={validateSchema(schema)}>
            {formikProps => (
              <RegisterFormNavigation
                step={step}
                formikProps={formikProps}
                isReviewing={isReviewing}
                getToReviewStep={getToReviewStep}
                nextStep={nextStep}
                previousStep={previousStep}
                navigateFormDuringReview={navigateFormDuringReview}
                handleRegister={handleRegister}
                registerLoading={registerLoading}
              />
            )}
          </Formik>
        </View>
      </SafeAreaView>
    </DismissKeyboard>
  );
};

export default Register;
