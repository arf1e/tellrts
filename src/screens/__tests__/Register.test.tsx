import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import {Alert} from 'react-native';
import {act} from 'react-test-renderer';
import {renderWithProviders} from '../../utils/test-utils';
import Register from '../Register';
import {REGISTER_MUTATION} from '../Register/Register.graphql';

const testUserRegisterData = {
  name: 'Egor',
  birthday: '16-10-1997',
  sex: true,
  photo: 'test-photo',
  countryCode: 'FI',
  cityId: 'testingCityId',
  password: 'corr3ctPassword',
  email: 'testinguser@tellr.dating',
};

const RegisterNewUser = () => (
  <Register
    route={{name: 'Signup', params: {email: testUserRegisterData.email}}}
    initialFormikTestingValues={{
      countryCode: testUserRegisterData.countryCode,
      countryTitle: 'Finland',
      cityId: testUserRegisterData.cityId,
      cityTitle: 'Lappeenranta',
    }}
  />
);
const nextStepTestId = 'Register.ControlPanel.next';
const previousStepTestId = 'Register.ControlPanel.back';

test('Register Screen Renders', () => {
  const view = render(renderWithProviders(<RegisterNewUser />));
  expect(view.toJSON()).toMatchSnapshot();
});

test('Pressing back on the first step causes modal to open', () => {
  jest.spyOn(Alert, 'alert');
  const view = render(renderWithProviders(<RegisterNewUser />));

  const nameTitle = view.getByText('register.name.title');
  expect(nameTitle).not.toBeNull();
  const sexTitle = view.queryByText('register.sex.title');
  expect(sexTitle).toBeNull();

  const backButton = view.getByTestId(previousStepTestId);
  fireEvent.press(backButton);
  expect(Alert.alert).toHaveBeenCalled();
});

test('Pressing next without value causes no effect', () => {
  const view = render(renderWithProviders(<RegisterNewUser />));
  const nextButton = view.getByTestId(nextStepTestId);
  const nameTitle = view.getByText('register.name.title');
  expect(nameTitle).not.toBeNull();
  fireEvent.press(nextButton);
  expect(nameTitle).not.toBeNull();
});

test('Next step becomes available after name input', async () => {
  const view = render(
    renderWithProviders(<RegisterNewUser />, {
      mocks: [
        {
          request: {
            query: REGISTER_MUTATION,
            variables: testUserRegisterData,
          },
          result: {
            data: {
              createAccount: {
                ok: true,
                error: 'woah',
                token: 'jwt-test-token',
              },
            },
          },
        },
      ],
    }),
  );
  const nameInput = await view.findByPlaceholderText(
    'register.name.fieldPlaceholder',
  );
  const nameTitle = view.getByText('register.name.title');
  expect(nameTitle).toBeOnTheScreen();
  const nextButton = view.getByTestId(nextStepTestId);
  await act(async () => {
    await fireEvent.changeText(nameInput, testUserRegisterData.name);
  });
  fireEvent.press(nextButton);
  expect(nameTitle).not.toBeOnTheScreen();
  const sexTitle = view.getByText('register.sex.title');
  expect(sexTitle).toBeOnTheScreen();
});

test('Register process', async () => {
  const view = render(renderWithProviders(<RegisterNewUser />));
  const nameTitle = view.getByText('register.name.title');
  expect(nameTitle).toBeOnTheScreen();
  const nameInput = await view.findByPlaceholderText(
    'register.name.fieldPlaceholder',
  );
  await act(async () => {
    await fireEvent.changeText(nameInput, testUserRegisterData.name);
  });
  fireEvent.press(view.getByTestId(nextStepTestId));
  const maleButton = view.getByText('register.sex.male');
  const femaleButton = view.getByText('register.sex.female');
  await act(async () => {
    await fireEvent.press(maleButton);
    await fireEvent.press(femaleButton);
    await fireEvent.press(maleButton);
  });
  fireEvent.press(view.getByTestId(nextStepTestId));
  const birthdayTitle = view.getByText('register.birthday.title');
  expect(birthdayTitle).toBeOnTheScreen();
  const birthdayInput = view.getByTestId('Register.birthday.BirthdayInput');
  await act(async () => {
    await fireEvent.changeText(birthdayInput, 'asdasdfafsa');
  });
  fireEvent.press(view.getByTestId(nextStepTestId));
  expect(birthdayTitle).toBeOnTheScreen();
  expect(birthdayInput).toBeOnTheScreen();
  await act(async () => {
    await fireEvent.changeText(birthdayInput, testUserRegisterData.birthday);
  });
  fireEvent.press(view.getByTestId(nextStepTestId));
  expect(birthdayTitle).not.toBeOnTheScreen();
  const photoTitle = view.getByText('register.photo.title');
  expect(photoTitle).toBeOnTheScreen();
  const cameraBtn = view.getByText('register.photo.camera');
  await act(async () => {
    await fireEvent.press(cameraBtn);
  });
  const takeAnotherPhoto = view.getByTestId('Register.Photo.TakeAnother');
  expect(view.getByTestId('Register.Photo.TakeAnother')).toBeOnTheScreen();
  fireEvent.press(takeAnotherPhoto);
  const galleryBtn = view.getByText('register.photo.gallery');
  await act(async () => {
    await fireEvent.press(galleryBtn);
  });
  expect(view.getByTestId('Register.Photo.TakeAnother')).toBeOnTheScreen();
  fireEvent.press(view.getByTestId(nextStepTestId));

  expect(view.getByText('register.location.title')).toBeOnTheScreen();
  expect(view.getByText('Finland')).toBeOnTheScreen();
  expect(view.getByText('Lappeenranta')).toBeOnTheScreen();
  fireEvent.press(view.getByTestId(nextStepTestId));

  expect(view.getByText('register.password.title')).toBeOnTheScreen();
  const passwordField = view.getByTestId('Register.Password.Password');
  const passwordConfirmField = view.getByTestId(
    'Register.Password.PasswordConfirm',
  );

  const fillPasswords = async (password: string) => {
    await act(async () => {
      await fireEvent.changeText(passwordField, password);
      await fireEvent.changeText(passwordConfirmField, password);
    });
  };

  const pass1 = 'wrongPassword';
  await fillPasswords(pass1);

  fireEvent.press(view.getByTestId(nextStepTestId));
  expect(view.getByText('register.password.title')).toBeOnTheScreen();

  await fillPasswords(testUserRegisterData.password);

  fireEvent.press(view.getByTestId(nextStepTestId));
  expect(view.queryByText('register.password.title')).toBeNull();
  expect(view.getByText('register.check.title')).toBeOnTheScreen();
  expect(view.getByText('register.check.nameTitle')).toBeOnTheScreen();
  expect(view.getByText(testUserRegisterData.name)).toBeOnTheScreen();
  expect(view.getByText('register.check.dobTitle')).toBeOnTheScreen();
  expect(view.getByText(testUserRegisterData.birthday)).toBeOnTheScreen();
  expect(view.getByText('register.check.sexTitle')).toBeOnTheScreen();

  await act(async () => {
    await fireEvent.press(view.getByText(testUserRegisterData.name));
  });

  expect(view.getByText('register.name.title')).toBeOnTheScreen();
  expect(view.getByText('register.controls.check')).toBeOnTheScreen();

  await act(async () => {
    await fireEvent.press(view.getByText('register.controls.check'));
  });

  expect(view.getByText('register.check.title')).toBeOnTheScreen();
});
