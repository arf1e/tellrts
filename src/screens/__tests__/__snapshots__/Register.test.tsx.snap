// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Screen Register Screen Renders 1`] = `
Array [
  <RCTSafeAreaView
    accessible={true}
    emulateUnlessSupported={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View>
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
        style={Object {}}
      >
        <View>
          <View>
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  undefined,
                  null,
                ]
              }
            />
            <View
              style={
                Array [
                  undefined,
                  null,
                ]
              }
            />
            <View
              style={
                Array [
                  undefined,
                  null,
                ]
              }
            />
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  undefined,
                  null,
                ]
              }
            />
            <View
              style={
                Array [
                  undefined,
                  null,
                ]
              }
            />
          </View>
        </View>
        <View
          animatedStyle={
            Object {
              "value": Object {},
            }
          }
          collapsable={false}
          entering={
            FadeInDown {
              "build": [Function],
              "delayV": 240,
              "durationV": 240,
              "randomizeDelay": false,
            }
          }
          exiting={
            FadeOutDown {
              "build": [Function],
              "durationV": 240,
              "randomizeDelay": false,
            }
          }
        >
          <Text
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            register.name.title
          </Text>
          <Text
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            register.name.description
          </Text>
          <View>
            <Text
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              register.name.fieldTitle
            </Text>
            <TextInput
              animatedStyle={
                Object {
                  "value": Object {
                    "borderColor": "rgba(188, 216, 221, 1)",
                  },
                }
              }
              collapsable={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="register.name.fieldPlaceholder"
              style={
                Object {
                  "borderColor": "rgba(188, 216, 221, 1)",
                }
              }
              testID="Field.TextInput"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              null,
            ]
          }
        >
          <View
            accessible={true}
            animatedStyle={
              Object {
                "value": Object {
                  "borderColor": "rgba(22, 147, 165, 1)",
                  "borderWidth": 1,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                },
              }
            }
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "rgba(22, 147, 165, 1)",
                "borderWidth": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
            testID="Register.ControlPanel.back"
          >
            <Text
              animatedStyle={
                Object {
                  "value": Object {
                    "color": "rgba(22, 147, 165, 1)",
                  },
                }
              }
              collapsable={false}
              style={
                Object {
                  "color": "rgba(22, 147, 165, 1)",
                }
              }
            >
              register.controls.back
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            animatedStyle={
              Object {
                "value": Object {
                  "backgroundColor": "rgba(22, 147, 165, 1)",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                },
              }
            }
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(22, 147, 165, 1)",
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
            testID="Register.ControlPanel.next"
          >
            <Text
              animatedStyle={
                Object {
                  "value": Object {
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  },
                }
              }
              collapsable={false}
              style={
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              register.controls.next
            </Text>
            <View
              animatedStyle={
                Object {
                  "value": Object {
                    "display": "none",
                    "opacity": 0,
                    "position": "absolute",
                    "transform": Array [
                      Object {
                        "translateY": 20,
                      },
                    ],
                  },
                }
              }
              collapsable={false}
              style={
                Object {
                  "display": "none",
                  "opacity": 0,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateY": 20,
                    },
                  ],
                }
              }
            >
              <ActivityIndicator
                color="#FDFFFF"
                size={24}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>,
  " ",
]
`;
